Podverse/Playlist.swift

```diff
@@ -222,6 +227,151 @@ class Playlist {
+        var body:[String: Any] = [:]
+        
+        if let title = title {
+            body["title"] = title
+        }
+        
+        if let itemsOrder = itemsOrder {
+            body["itemsOrder"] = itemsOrder
+        }
+        
+        return body
+    }
+    
+    static func updatePlaylistOnServer(id: String, title:String?, itemsOrder:[String]? = [], completion: @escaping (Bool) -> Void) {
+        if let url = URL(string: BASE_URL + "playlists/" + id) {
+            
+            var request = URLRequest(url: url, cachePolicy: NSURLRequest.CachePolicy.reloadIgnoringLocalAndRemoteCacheData, timeoutInterval: 20)
+            
+            request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+            
+            request.httpMethod = "PUT"
+            
+            if let idToken = UserDefaults.standard.string(forKey: "idToken") {
+                request.setValue(idToken, forHTTPHeaderField: "Authorization")
+            }
+            
+            let putBody = self.convertToPlaylistUpdateBody(title: title, itemsOrder: itemsOrder)
+            
+            do {
+                request.httpBody = try JSONSerialization.data(withJSONObject: putBody, options: JSONSerialization.WritingOptions())
+                
+                showNetworkActivityIndicator()
+                
+                let task = URLSession.shared.dataTask(with: request) { data, response, error in
+                    hideNetworkActivityIndicator()
+                    
+                    guard error == nil else {
+                        DispatchQueue.main.async {
+                            completion(false)
+                        }
+                        return
+                    }
+                    
+                    DispatchQueue.main.async {
+                        completion(true)
+                    }
+                }
+                
+                task.resume()
+                
+            } catch {
+                print(error)
+                DispatchQueue.main.async {
+                    completion(false)
+                }
+            }
+            
+        }
+    }
+    
+    static func subscribeToPlaylistOnServer(id:String, completion: @escaping (Bool) -> Void) {
+        if let url = URL(string: BASE_URL + "playlists/subscribe/" + id), let idToken = UserDefaults.standard.string(forKey: "idToken") {
+            showNetworkActivityIndicator()
+            
+            let request = NSMutableURLRequest(url: url, cachePolicy: NSURLRequest.CachePolicy.reloadIgnoringLocalAndRemoteCacheData, timeoutInterval: 60)
+            
+            request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+            request.setValue(idToken, forHTTPHeaderField: "Authorization")
+            
+            request.httpMethod = "POST"
+            
+            let task = URLSession.shared.dataTask(with: request as URLRequest) { data, response, error in
```

Comment:
There is so much repetition of this URLSession.shared.dataTask call I'm almost inclined to maybe create a helper function for it and replace it everywhere in this file. something like

```
func performDataTask(_ request:UrlRequest, _ completion:((completed)->()) {
    return URLSession.shared.dataTask(with: request) { data, response, error in
    hideNetworkActivityIndicator()
                guard error == nil else {
                    print("Error: \(error?.localizedDescription ?? "Unknown Error")")
                    DispatchQueue.main.async {
                        completion(false)
                    }
                    return
                }
                
                completion(true)
            }
}
```

and then call it like this maybe:

`performDataTask(request, completion).resume()`

Reply:
True, this is a great idea, however I'm struggling to figure out how to handle the different behaviors in the task completion blocks within this file...

Moreover, variants of this set request and dataTask pattern is used in MediaRef, PlayerHistoryItem, Episode, Podcast, SearchPodcast, PVAuth, and PVSubscriber...how about we handle rewriting these in a separate PR, and do them all at once?

Reply:
Added an issue for this https://github.com/podverse/podverse-ios/issues/187